### PR TITLE
SAM-1408

### PR DESCRIPTION
**Pull request by gillmanjc**
I am working on this Jira. I have manually added in the changes provided in the attached patch to a Sakai trunk instance.
The patch however was created from an older sakai-10.x branch, which compared to trunk had some conflicts etc. 
After setting up a test Scenario to test the patch changes, I ran into an issue with authorSettings.jsp not rendering successfully. 
I tried to investigate what the cause might be, because it does not seem like the added changes is the culprit here. I have created an PR.

### CODE REVIEW

File: samigo/samigo-app/src/webapp/js/extendedTime.js [hunk 1]
```diff
@@ -0,0 +1,208 @@
...
+// This is the method called from authorSettings.jsp to set up all the extended time stuff
+function extendedTimeInitialize() {
+	addAllExtendedTimeItems();
+	allExtendedTimeEntries = document.getElementById("assessmentSettingsAction\:xt1").value.split("^");
+
+	for (var itemNum = 1; itemNum <= MAXITEMS; itemNum++) { 
+		copyListValuesForExtTime(itemNum);
+	}
+
+	var itemNum = 1
```
Comment (steveswinsburg) [inline samigo/samigo-app/src/webapp/js/extendedTime.js]: missing semi colon at end of this line

File: samigo/samigo-app/src/webapp/js/extendedTime.js [hunk 2]
```diff
@@ -0,0 +1,208 @@
...
+		activeExtTimeEntries++; 
+		fullExtendedTimeString = allExtendedTimeEntries[itemNum-1].split("|");
+		initializeExtTimeValues(fullExtendedTimeString,itemNum);
+		showActiveExtTimeEntities(fullExtendedTimeString,itemNum);
+		itemNum++;
+			
+	} while(itemNum < allExtendedTimeEntries.length);
+}
+
+// We secretly store the extended time info in a hidden JSF input field. Whenever a user edits an
```
Comment (steveswinsburg) [inline samigo/samigo-app/src/webapp/js/extendedTime.js]: Had to giggle at the 'secretly'. In a file that is sent to the client :+1:

File: samigo/samigo-app/src/webapp/js/extendedTime.js [hunk 3]
```diff
@@ -0,0 +1,208 @@
...
+			"</td></tr><tr><td>"+
+			"Retract Date</td><td>"+
+			"<input type=\"text\" size=\"25\"  id=\"xt_retract"+itemNum+"\">&nbsp;"+
+			"<img id=\"_datePickerPop_xt_retract"+itemNum+"\" height=\"16\" border=\"0\" width=\"16\" alt=\"Click Here to Pick Date\" src=\"/samigo-app/images/calendar/cal.gif\" style=\"cursor:pointer;\" onclick=\"javascript:var cal06670970441558088 = new calendar2(document.getElementById('xt_retract"+itemNum+"'), cal_gen_date2_md, cal_prs_date2_md);cal06670970441558088.year_scroll = true;cal06670970441558088.time_comp = true;cal06670970441558088.popup('','/samigo-app/html/');\">"+
+			"</td></tr></table>"+
+			"</div> <!--end dates -->"+
+			"<hr width=\"450\"  align=\"left\">"+
+			"</div> <!--end xt"+itemNum+" -->";
```
Comment (steveswinsburg) [inline samigo/samigo-app/src/webapp/js/extendedTime.js]: This all seems very fragile :(

**Comment by steveswinsburg** [comment]
What is the issue with authorSettings rendering? Its probably all of the javascript it includes. Does the browser JS console give any hints? Maybe step through the JS methods and see whats happening there.

**Comment by gillmanjc** [comment]
Hi Steve
I have made some additional changes which seems to be working fine now. Had some page rendering issues with the patch, etc. Please have a look again and let me know.
Will continue with the other Jiras!

**Comment by steveswinsburg** [comment]
Hi Jaco,
So in your environment is this all rendering fine now and the feature working well?
cheers,
Steve

**Comment by gillmanjc** [comment]
Hi Steve

Everything in the patch provided mentioned in the Jira has been pushed to this branch and is part of this PR. However I am experiencing issues when testing the functionality.

I managed to fix the Calendar popup to be able to choose a date which was not working when I applied the changes from the patch.

Here are my test results:
1) First time link click: "Add a Time Limit/Delivery Date Exception" it only changes to "Add another Time Limit/Delivery Date Exception."
2) Drop down fields in this view has no values for hrs & mins.
3) After selecting date values for "Available Date", "Due Date" & "Retract Date", I choose to save. When I then go back to view the Settings, these date values aren't populated.

I would like to get in contact with the developers who contributed the patch, because it seems like there is still some work to be done to get the Jira resolved other than that of the patch itself. Do you think its best if I copy this comment to the Jira?

Kind Regards
Jaco

**Comment by steveswinsburg** [comment]
Hi Jaco,
Yes please put this into the Jira itself. It would probably be good to chat to Paul Dagnall from Dayton University, I believe he worked on this one.

cheers,
Steve

**Comment by gillmanjc** [comment]
Hi David

Thank you very much for looking at this Jira and your email reply. 

The issue you mentioned about the error you get in extendedTime.js, was one that I also had. I have changed this value to extendedTimeTarget, and now see changes I didn't see when I tested the patch originally. 

The dropdown lists now gets populated and it values that I select does get saved. I unfortunately haven't had time to test the actual extended hours limitations being configured as part of the new this new patch contribution functionality. 

@Steve, can you please comment on how David can contribute his changes? I think maybe send a diff or patch file to Steve?

**Comment by steveswinsburg** [comment]
How big is the change? Can you just attach it here and then it can be incorporated? I don't have the branch locally so I wouldn't be able to integrate it.
Is that fix everything that you need? 

cheers,
Steve

**Comment by gillmanjc** [comment]
:) the change was a one line change! (see Updated variable value Id - SAM-1408) I have already committed as part of this PR.

**Comment by steveswinsburg** [comment]
Ok I'm ready to pull this down and test it out if you think all code is in
for this one? Let me know and I'll get that done.

**Comment by gillmanjc** [comment]
Hi Steve, yes please go ahead and pull this down as it is.

**Comment by steveswinsburg** [comment]
Hey Jaco, receiving a compilation error:
````
 Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.3:compile (default-compile) on project samigo-app: Compilation failure
[ERROR] /home/travis/build/sakaiproject/sakai/samigo/samigo-app/src/java/org/sakaiproject/tool/assessment/util/ExtendedTimeService.java:[9,36] cannot find symbol
[ERROR] symbol:   class AuthzGroupService
[ERROR] location: package org.sakaiproject.authz.cover
````

**Comment by gillmanjc** [comment]
Hi Steve
I have switched my workspace branch to SAM-1408 and rebuild a few times, mine does compile just fine. The new class ExtendedTimeService does make use of the deprecated AuthzGroupService but it does however ships with the sakai-kernel-api dependency version 11-SNAPSHOT.

**Comment by steveswinsburg** [comment]
Running the build again.

**Comment by steveswinsburg** [comment]
It's still failing :(

**Comment by ncaidin** [comment]
Fyi, I've assigned the Jira to Jaco G. and set the status to "In Progress" on the ticket.

**Comment by ern** [comment]
Need to use rebase instead of merge

**Comment by steveswinsburg** [comment]
Ok It's now building However the settings page layout is a bit screwed up. The headings *About*, *Availability*, *Grading* etc are all nested under the top level heading. This is not the case in nightly, they are all separate sections. Could you have a look at the markup?
screen shot 2015-09-28 at 12 24 21 pm
screen shot 2015-09-28 at 12 25 07 pm

**Comment by gillmanjc** [comment]
Hi Steve

I have committed some more changes after I updated my local workspace with master from trunk. Fixed some issues that weren't in the original patch contribution and certainly not when I tested.

Hopefully this patch contribution now plays along! Please retest and let me know..

**Comment by ottenhoff** [comment]
41 commits on this PR :(

Can someone rebase?

**Comment by steveswinsburg** [comment]
Closing this. See #1142 for a new PR.

**Pull request closed without merging**